internal/connectors/plugins/public/mangopay/client/bank_accounts.go

```diff
@@ -130,16 +128,11 @@ func (c *Client) createBankAccount(ctx context.Context, endpoint string, req any
 	httpReq.Header.Set("Content-Type", "application/json")
 
 	var bankAccount BankAccount
-	_, err = c.httpClient.Do(httpReq, &bankAccount, nil)
-	switch err {
-	case nil:
-		return &bankAccount, nil
-	case httpwrapper.ErrStatusCodeUnexpected:
-		// Never retry bank account creation
-		// TODO(polo): retry ?
-		return nil, err
+	statusCode, err := c.httpClient.Do(httpReq, &bankAccount, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create bank account, got code %d: %w", statusCode, err)
```

Comment:
can this used the sharedError pkg ?

Reply:
Are you referring to a package in particular, or do you just mean to create a shared package which helps format such errors?

Reply:
I updated the mangopay client to use `go-libs/errorsutils` to handle the status code as you suggested.